docs/writing-a-profile.md

```diff
@@ -0,0 +1,166 @@
+# Writing a profile
+
+**Per endpoint, `hafas-client` has an endpoint-specific customisation called *profile*** which may for example do the following:
+
+- handle the additional requirements of the endpoint (e.g. authentication),
+- extract additional information from the data provided by the endpoint,
+- guard against triggering bugs of certain endpoints (e.g. time limits).
+
+This guide is about writing such a profile. If you just want to use an already supported endpoint, refer to the [API documentation](readme.md) instead.
+
+*Note*: **If you get stuck, ask for help by [creating an issue](https://github.com/derhuerst/hafas-client/issues/new)!** I want to help people expand the scope of this library.
+
+## 0. How do the profiles work?
+
+A profile contains of three things:
+
+- **mandatory details about the HAFAS endpoint**
+	- `endpoint`: The protocol, host and path of the endpoint.
+	- `locale`: The [BCP 47](https://en.wikipedia.org/wiki/IETF_language_tag) [locale](https://en.wikipedia.org/wiki/Locale_(computer_software)) of your endpoint (or the area that your endpoint covers).
+	- `timezone`: An [IANA-time-zone](https://www.iana.org/time-zones)-compatible [timezone](https://en.wikipedia.org/wiki/Time_zone) of your endpoint.
+- **flags indicating that features are supported by the endpoint** – e.g. `journeyRef`
+- **methods overriding the [default profile](../lib/default-profile.js)**
+
+As an example, let's say we have an [Austrian](https://en.wikipedia.org/wiki/Austria) endpoint:
+
+```js
+const myProfile = {
+	endpoint: 'https://example.org/bin/mgate.exe',
+	locale: 'de-AT',
+	timezone: 'Europe/Vienna'
+}
+```
+
+If you pass this profile into `hafas-client`, the `parseLine` method will override [the default one](../parse/line.js).
```

Comment:
> If you pass this profile into `hafas-client`…

This is a bit confusing since `myProfile` doesn't contain any `parseLine` method (yet)